### ISSUE TITLE
'with_community.general.filetree' is not a valid attribute for a Task

### ISSUE THREAD

**Issue by 4wk-** [ansible/ansible-lint#4304]
##### SUMMARY

`ansible-lint` is complaining about attribute `with_community.general.filetree` not being valid. However, it's in [Ansible official documentation](https://docs.ansible.com/ansible/latest/collections/community/general/filetree_lookup.html). cf examples.

![image](https://github.com/user-attachments/assets/ea4b14b6-e944-4e23-97c3-29c9fd5e9fde)

Am I missing something?

##### ISSUE TYPE

- Bug Report

##### OS / ENVIRONMENT

Debian 12.6 in wsl2

```console (paste below)
$ ansible-lint --version
ansible-lint 24.7.0 using ansible-core:2.17.2 ansible-compat:24.7.0 ruamel-yaml:0.18.6 ruamel-yaml-clib:0.2.8
```

- ansible installation method: `pipx`
- ansible-lint installation method: `pipx`

##### STEPS TO REPRODUCE

```yml
---
- name: testing123
  hosts: all
  become: yes
  gather_facts: no
  tasks:

    - name: Testing filetree
      ansible.builtin.template:
        src: "{{ item.path }}"
        dest: "/tmp/"
        owner: root
        group: root
        mode: '644'
      with_community.general.filetree: "templates/"
      when: item.state == 'file'
```

##### DESIRED BEHAVIOR
`ansible-lint` should not complain about `with_community.general.filetree`.

##### ACTUAL BEHAVIOR

```console
$ ansible-lint zplay.yml
WARNING  Listing 1 violation(s) that are fatal
syntax-check[specific]: 'with_community.general.filetree' is not a valid attribute for a Task
zplay.yml:11:7


                  Rule Violation Summary
 count tag                    profile rule associated tags
     1 syntax-check[specific] min     core, unskippable

Failed: 1 failure(s), 0 warning(s) on 1 files. Profile 'production' was required.
```

**Comment by cavcrosby** [comment]
I was able to reproduce this when deleting the `community.general` collection from all directories within Ansible's `collections_path`. Can you check to see if the collection is installed, and that the `collections_path` configuration is set appropriately to find the collection?

**Comment by leonardgerardatomicmachinescom** [comment]
Is the linter dependent on the specific ansible that is installed in the system? (we run the linter in CI, which doesn't install ansible)

**Comment by cavcrosby** [comment]
@leonardgerardatomicmachinescom I believe `ansible-lint` will work with versions of `ansible`/ `ansible-core` that meet the version constraint `ansible-core>=2.13.0`. I'm basing this on the "requirements.txt" file used to specify runtime dependencies, as shown below.

https://github.com/ansible/ansible-lint/blob/7b9749373a4282d2fe6fc15a6ce3e65d41f7efae/.config/requirements.in#L3

**Comment by leonardgerardatomicmachinescom** [comment]
I guess I did not expect the lint rules to be dependent on the installed ansible-core, but solely on the installed ansible-lint version.
We will have to pay extra attention to have the same ansible version in our CI linting the ansible and in our deployment system (semaphore).

You can close this issue, since it is probably "as designed".
